R/setup_forested.R

```diff
@@ -0,0 +1,129 @@
+library(sf)
+library(tidymodels)
+library(spatialsample)
+library(tidysdm)
+library(forested)
+
+# ------------------------------------------------------------------------------
+
+tidymodels_prefer()
+theme_set(theme_bw())
+options(pillar.advice = FALSE, pillar.min_title_chars = Inf)
+
+# ------------------------------------------------------------------------------
+
+# These computations take a while so we do them in batch mode then load the 
+# results in later chapters
+
+# ------------------------------------------------------------------------------
+# We remove some of the existing algorithmic data already there. 
+
+for_analysis <-
+  forested %>%
+  rename(class = forested) %>%
+  select(-tree_no_tree, -land_type, -canopy_cover) 
+
+# ------------------------------------------------------------------------------
+# Convert the lon/lat to sf geometry format
+
+forested_sf <-
+  for_analysis %>%
+  st_as_sf(coords = c("lon","lat"), crs = st_crs("EPSG:4326"))
+
+
+# ------------------------------------------------------------------------------
+# Conduct the initial split using block sampling and buffering
+
+set.seed(318)
+forested_split <-
+  spatial_initial_split(forested_sf, prop = 1 / 5, spatial_block_cv,
+                        buffer = 80 * 80, method = "continuous",
+                        n = 25, square = FALSE)
+
+forested_sf_train <- training(forested_split)
+forested_sf_test  <- testing(forested_split)
+
+# ------------------------------------------------------------------------------
+# Make a data frame to use for plotting
+
+split_groups <- tibble(.row = 1:nrow(forested_sf), group = "buffer")
+split_groups$group[forested_split$in_id] <- "training"
+split_groups$group[forested_split$out_id] <- "testing"
+
+forested_split_info <- 
+  for_analysis %>% 
+  add_rowindex() %>% 
+  full_join(split_groups, by = ".row") %>% 
+  mutate(
+    group_col = if_else(group == "training", "#E7298A", "#7570B3"),
+    group_col = if_else(group == "buffer", "#000000", group_col)
+  )
+
+# ------------------------------------------------------------------------------
+# Resample the training set
+
+set.seed(670)
+forested_rs <- spatial_block_cv(
+  forested_sf_train,
+  v = 10,
+  buffer = 80 * 80,
+  method = "continuous",
+  n = 25,
+  square = FALSE
+)
+
+# ------------------------------------------------------------------------------
+# re-calculate geocodes and strip off geometry column
+
+re_geocode <- function(x) {
```

Comment:
We'll be using this data in later chapters and will use standard ML tools. Since the `geometry` is sticky, we delete it from the data/resamples and add the longitude and latitude back in. 

Perhaps there is an easier way to do this>